app/assets/javascripts/discourse/app/routes/application.js

```diff
@@ -278,11 +278,12 @@ export default class ApplicationRoute extends DiscourseRoute {
 
   handleShowCreateAccount(createAccountProps) {
     if (this.siteSettings.enable_discourse_connect) {
-      const returnPath = encodeURIComponent(window.location.pathname);
+      const returnPath = cookie("destination_url")
```

Comment:
When Discourse Connect is enabled, we have a slightly different behavior for the `return path` when there's a `destination_url` cookie. I applied the same behavior when going through the signup process too.

This is the 2nd reason why this PR is a draft as I'm not 100% sure the impact this change has 🤔